### ISSUE TITLE
Has anyone experienced this error: malloc(): memory corruption with extrinsic_calib?

### ISSUE THREAD

**Issue by stanathong** [hengli/camodocal#46]
Hello,

I have set up an Ubuntu 14.04 with OpenCV3.0 and CUDA. Using the previous committed source code, CamOdoCal was built successfully. However, when I ran it, it showed this error:

.# INFO: Initializing... 
.# INFO: Initialization finished!
GPS: lat=53.9487, lon=-1.04606, alt=0, qx=0, qy=0, qz=0.913545, qw=-0.406737 [9953]
GPS: lat=53.9487, lon=-1.04606, alt=0, qx=0, qy=0, qz=0.913545, qw=-0.406737 [10015]
GPS: lat=53.9487, lon=-1.04606, alt=0, qx=0, qy=0, qz=0.913545, qw=-0.406737 [10078]
GPS: lat=53.9487, lon=-1.04606, alt=0, qx=0, qy=0, qz=0.913545, qw=-0.406737 [10156]
GPS: lat=53.9487, lon=-1.04606, alt=0, qx=0, qy=0, qz=0.913545, qw=-0.406737 [10218]
GPS: lat=53.9487, lon=-1.04606, alt=0, qx=0, qy=0, qz=0.913545, qw=-0.406737 [10280]
IMG: 10218 -> /home/supannee/src/camodocal-master/gaist_data/sample_test//frames_0/Front_154.jpg
.# INFO: Running camera-odometry calibration for each of the 3 cameras.
*** Error in `bin/extrinsic_calib': malloc(): memory corruption: 0x00007fc416238f20 ***
LOC: 5979583.339446 628229.702183 0.000000

I had no error running the same dataset on a Virtual Machine Ubuntu (without CUDA). 

Just in case, anyone has found this error before and know how to fix this. Thank you very much.

**Comment by ahundt** [comment]
Unfortunately some situations may not be tested, I did my best to port it but there may be some bugs and I didn't have a way to test. Check in the open/closed issues/pull requests. I think someone mentioned that the gpu objects aren't being created 100% correctly, perhaps that is related?

**Comment by stanathong** [comment]
Hi @ahundt, thanks for getting back to me.

I have wiped out my ubuntu and tried the whole process again. I found out that without installing nvidia-cuda-toolkit the problem is solved. I didn't see that reported error any more.

However, when start running it, with SURF_GPU_DETECTOR and SURF_GPU_DESCRIPTOR, the program broke when calling SurfGPU::knnMatch even there were non-empty keypoints. CamOdoCal with GPU seems to fail since the visual odometry stage.

While for the CamOdoCal without GPU, it could pass through until the bundle adjustment process but failed in Stage 3 - find local inter-camera 3D-3D correspondences because the program was trying to use SurfGPU, as reported in my previous issue https://github.com/hengli/camodocal/issues/43.

Does anyone has any completed code with working either with GPU or without GPU?? Thank you very much.
